### PR TITLE
perf(dependency): update dependency

### PR DESCRIPTION
**Pull request by DR-Univer**
## What's updated?
- [x] **IMPORTANT**: Data validation and conditional formatting, register custom formulas, support passing `f` and `ranges`, no need to make offsets in the plugin in advance @weird94 @Gggpound  Please check
- [x] Formula supports configuration of calculation mode
    ```ts
    enum CalculationMode {
        /**
         * Force calculation of all formulas
         */
        FORCED,
    
        /**
         * Partial calculation, only cells with formulas but no v values are calculated
         */
        WHEN_EMPTY,
    
        /**
         * All formulas are not calculated
         */
        NO_CALCULATION,
    }
    ```
- [x] Add current task description to the formula progress bar
    企业微信截图_17302907787461
- [x] Remove `console.warn` during formula calculation time, use `log.service` instead
    image

## Pull Request Checklist

- [x] Related tickets or issues have been linked in the PR description (or missing issue).
- [x] [Naming convention](https://github.com/dream-num/univer/blob/dev/docs/NAMING_CONVENTION.md) is followed (**do please** check it especially when you created new plugins, commands and resources).
- [x] Unit tests have been added for the changes (if applicable).
- [x] Breaking changes have been documented (or no breaking changes introduced in this PR).

### CODE REVIEW

**Comment by zhaolixin7** [comment]
@DR-Univer 1、选中行填充公式，表现：一直显示进度条不计算，控制台报错（snapshot太大上传不了）
企业微信20241101-155944@2x

**Comment by zhaolixin7** [comment]
@DR-Univer  条件格式类型时公式时图标颜色显示不对
image
[2024_11_1 17_04_39 snapshot.json](https://github.com/user-attachments/files/17598069/2024_11_1.17_04_39.snapshot.json)

**Comment by zhaolixin7** [comment]
新建数据验证一直显示公式进度条
image
cc. @DR-Univer @weird94 
[2024_11_1 19_26_31 video.webm](https://github.com/user-attachments/assets/85ababc7-1aa1-492f-85e1-53bfd2ec07c7)

**Comment by zhaolixin7** [comment]
选中A1:AD4拖拽填充撤销后再恢复 表现：报错不再计算

https://github.com/user-attachments/assets/3680767a-c817-43a7-b93a-fb9f6aa0a99b

**Comment by Dushusir** [comment]
> 新建数据验证一直显示公式进度条 image
> 
>  2024_11_1.19_26_31.video.webm

fixed

File: packages/engine-formula/src/services/calculate-formula.service.ts [hunk 1]
```diff
@@ -246,10 +249,16 @@ export class CalculateFormulaService extends Disposable {
...
         const config = this._configService.getConfig(PLUGIN_CONFIG_KEY) as IUniverEngineFormulaConfig;
         const intervalCount = config?.intervalCount || DEFAULT_INTERVAL_COUNT;
-
-        for (let i = 0, len = treeList.length; i < len; i++) {
-            const tree = treeList[i];
-            const nodeData = tree.nodeData;
+        let i = 0;
+        const treeCount = treeList.length;
+        while (treeList.length > 0) {
+            const tree = treeList.pop()!;
```
Comment (VicKun4937) [inline packages/engine-formula/src/services/calculate-formula.service.ts:255]: use if instead "!"

**Comment by zhaolixin7** [comment]
1、数据验证自定义公式结果不对，不符合区域报错
2、修改自定义公式偶现2次进度条不消失的情况 cc @Dushusir image

**Comment by zhaolixin7** [comment]
@Dushusir 在条件格式区域输入小于-1的就会变成文本
复现：[2024_11_2 14_39_03 snapshot.json](https://github.com/user-attachments/files/17605969/2024_11_2.14_39_03.snapshot.json)
表现：
[2024_11_2 14_35_18 video.webm](https://github.com/user-attachments/assets/816afd0c-4c45-423d-a18a-388b71a15532)

**Comment by Dushusir** [comment]
> @DR-Univer @Gggpound 1、色阶/图标集最大最小值为公式时颜色显示不对 image image
> 
> [2024_11_1 17_04_39 snapshot.json](https://github.com/user-attachments/files/17598069/2024_11_1.17_04_39.snapshot.json) 2、数据条没显示 [2024_11_1 19_04_46 snapshot.json](https://github.com/user-attachments/files/17598935/2024_11_1.19_04_46.snapshot.json)
> 
> image

Fixed

企业微信截图_1730533421529

企业微信截图_173053341830

企业微信截图_17305335214421

**Comment by Dushusir** [comment]
> 选中A1:AD4拖拽填充撤销后再恢复 表现：报错不再计算
> 
>  2024-11-01.20.31.25.mov

Duplicate of https://github.com/dream-num/univer/pull/3875#issuecomment-2451484311

**Pull request closed without merging**